source/sundials/arkode.cc

```diff
@@ -239,9 +246,20 @@ namespace SUNDIALS
     int status;
     (void)status;
 
-#  if DEAL_II_SUNDIALS_VERSION_GTE(6, 0, 0)
+#  if DEAL_II_SUNDIALS_VERSION_GTE(7, 0, 0)
+    status = SUNContext_Free(&arkode_ctx);
+    AssertARKode(status);
```

Comment:
These two lines are duplicated between the if and else branches. You can move them out of the block.

Reply:
These two lines should not be present with SUNDIALS < 6.0.0.

The conditional actually consists of an `#if` and `#elif` branch. Thus, I would like to keep these two lines there, to avoid a nested conditional.